include/stdexec/execution.hpp

```diff
@@ -5485,6 +5485,203 @@ namespace stdexec {
   using __when_all::transfer_when_all_with_variant_t;
   inline constexpr transfer_when_all_with_variant_t transfer_when_all_with_variant{};
 
+  namespace __when_any_ {
+  template <class _BaseEnv>
+  using __env_t =
+      __make_env_t<_BaseEnv, __with<get_stop_token_t, in_place_stop_token>>;
+
+  template <class _Receiver, class _Op> struct __receiver_id {
+    class __t {
+    public:
+      using __id = __receiver_id;
+
+      explicit __t(_Op* __op) noexcept : __op_{__op} {}
+
+    private:
+      _Op* __op_;
+
+      template <class _CPO, class... _Args>
+      void notify(_CPO, _Args&&... __args) noexcept {
+        __op_->notify(_CPO{}, ((_Args &&) __args)...);
+      }
+
+      __env_t<env_of_t<_Receiver>> get_env() const noexcept {
+        using with_token = __with<get_stop_token_t, in_place_stop_token>;
+        auto __token = with_token{__op_->__stop_source_.get_token()};
+        return __make_env(stdexec::get_env(__op_->__receiver_), (with_token &&) __token);
+      }
```

Comment:
I'm curious why you made this a member function instead of just doing this in the `get_env_t` `tag_invoke` overload below.